containers/ecr-viewer/src/app/api/services/extended_database_repo.ts

```diff
@@ -0,0 +1,703 @@
+import { Kysely } from "kysely";
+
+import { db } from "./database";
+import {
+  Extended,
+  ExtendedECR,
+  NewExtendedECR,
+  ExtendedECRUpdate,
+  PatientAddress,
+  NewPatientAddress,
+  PatientAddressUpdate,
+  ECRLabs,
+  NewECRLabs,
+  ECRLabsUpdate,
+  ECRConditions,
+  NewECRConditions,
+  ECRConditionsUpdate,
+  ECRRuleSummaries,
+  NewECRRuleSummaries,
+  ECRRuleSummariesUpdate,
+} from "./extended_types";
+
+/**
+ * Finds an eICR by its ID
+ * @async
+ * @function findEcrById
+ * @param id - the ID of the ExtendedEcr being looked up
+ * @returns an eICR object
+ */
+export async function findExtendedEcrById(id: string | null) {
+  if (!id) {
+    return console.error("eICR ID is required.");
+  }
+  try {
+    return await (db as Kysely<Extended>)
+      .selectFrom("ecr_data")
+      .where("eICR_ID", "=", id)
+      .selectAll()
+      .executeTakeFirst();
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+/**
+ * Finds an eICR by its given criteria
+ * @async
+ * @function findExtendedEcr
+ * @param criteria - the Partial<ExtendedECR> filtering criteria
+ * @returns an eICR object
+ */
+export async function findExtendedEcr(criteria: Partial<ExtendedECR> | null) {
+  let query = (db as Kysely<Extended>).selectFrom("ecr_data");
+
+  if (!criteria || criteria === null) {
+    return console.error("eICR Criteria is required.");
+  }
```

Comment:
do we need this separate from the `findEcr`? or could we make it generic on the type of ECR and then only have one function?